packages/inventory/src/components/table/EntityTableToolbar.js

```diff
@@ -0,0 +1,267 @@
+import { fetchAllTags, clearFilters, entitiesLoading, toggleTagModal } from '../../redux/actions';
+import debounce from 'lodash/debounce';
+import flatMap from 'lodash/flatMap';
+import {
+    TagsModal,
+    mapGroups,
+    TEXT_FILTER,
+    reduceFilters,
+    TEXTUAL_CHIP,
+    STALE_CHIP,
+    REGISTERED_CHIP,
+    TAG_CHIP,
+    arrayToSelection,
+    loadSystems
+} from '../../shared';
+import { onDeleteFilter, onDeleteTag } from './helpers';
+import { useStalenessFilter, useTextFilter, useRegisteredWithFilter, useTagsFilter } from '../filters';
+
+const EntityTableToolbar = ({
+    total,
+    page,
+    perPage,
+    filterConfig,
+    hasItems,
+    children,
+    actionsConfig,
+    onRefresh,
+    hasCheckbox,
+    activeFiltersConfig,
+    showTags,
+    isLoaded,
+    items,
+    sortBy,
+    ...props
+}) => {
+    const dispatch = useDispatch();
+    const filters = useSelector(({ entities: { activeFilters } }) => activeFilters || []);
+    const loaded = useSelector(({ entities: { loaded } }) => hasItems && isLoaded !== undefined ? (isLoaded && loaded) : loaded);
+    const allTagsLoaded = useSelector(({ entities: { allTagsLoaded } }) => allTagsLoaded);
+    const allTags = useSelector(({ entities: { allTags } }) => allTags);
+    const additionalTagsCount = useSelector(({ entities: { additionalTagsCount } }) => additionalTagsCount);
+    const [ nameFilter, nameChip, textFilter, setTextFilter ] = useTextFilter('');
+    const [ stalenessFilter, stalenessChip, staleFilter, setStaleFilter ] = useStalenessFilter();
+    const [ registeredFilter, registeredChip, registeredWithFilter, setRegisteredWithFilter ] = useRegisteredWithFilter();
+    const [
+        tagsFilter,
+        tagsChip,
+        selectedTags,
+        setSelectedTags,
+        filterTagsBy
+    ] = useTagsFilter(allTags, allTagsLoaded, additionalTagsCount, () => dispatch(toggleTagModal(true)));
+    const onRefreshData = useCallback((options) => dispatch(loadSystems(options, showTags)));
+
+    const updateData = (config) => {
+        const params = {
+            items,
+            page,
+            per_page: perPage,
+            filters,
+            hasItems,
+            ...config
+        };
+        onRefresh ? onRefresh(params, (options) => {
+            dispatch(entitiesLoading());
+            onRefreshData({ ...params, ...options });
+        }) : onRefreshData(params);
+    };
+
+    const debouncedRefresh = useCallback(debounce((config) => updateData(config), 800), []);
+    const debounceGetAllTags = useCallback(debounce((config, options) => {
+        if (showTags && !hasItems) {
+            dispatch(fetchAllTags(config, options));
+        }
+    }, 800), []);
+
+    useEffect(() => {
+        const { textFilter, tagFilters, staleFilter, registeredWithFilter } = reduceFilters(filters);
+        debouncedRefresh();
+        ReactDOM.unstable_batchedUpdates(() => {
```

Comment:
Wouldn't it be safer to switch to `useReducer` than using experimental features? This can break with any patch/minor update of React. I know that the react devs stated that this should not change but they have also stated that any unstable method should be used with caution :man_shrugging: 

Considering that react-dom is an external dependency it can cause issues with applications with an older package version.

Reply:
The problem is that each hook is using custom state, however this can be removed as it was just to prevent muliple re-renders (at one point was firing multiple GET requests, but I've fixed that in other places).

Reply:
I think that this is a weak point and can cause unpredictable behavior in consumer apps. I would even prefer doing multiple re-renders over it.

Reply:
Yeah, good point. I am working on an enhancement to use reducer wih multiple hooks.